rest/handler.go

```diff
@@ -203,11 +203,24 @@ func newHandler(server *ServerContext, privs handlerPrivs, serverType serverType
 // ctx returns the request-scoped context for logging/cancellation.
 func (h *handler) ctx() context.Context {
 	if h.rqCtx == nil {
-		h.rqCtx = base.CorrelationIDLogCtx(h.rq.Context(), h.formatSerialNumber())
+		ctx := base.CorrelationIDLogCtx(h.rq.Context(), h.formatSerialNumber())
+		serverAddr, err := h.getServerAddr()
+		if err != nil {
```

Comment:
If this errors in prod, this will be empty string, I wonder if it makes sense to say "unable to determine" or something.

Reply:
In the actual value? I'm not sure it's worth the special case (audit logging code will parse this value to split host and port)

The scenario would be that we start up an API server, don't include it on the `ServerContext`'s `_httpServers` map for some reason.

I think we could probably get test coverage to ensure this is working correctly.

Reply:
This is fine to skip, I forgot that audit logging code also parses this and we handle there.